packages/templates/tests.spec.ts

```diff
@@ -0,0 +1,100 @@
+import fs from "fs";
+import path from "path";
+import { exec } from "child_process";
+
+jest.setTimeout(600000);
+
+describe("Templates", () => {
+  const rootDir = __dirname;
+
+  // Mapping of all project types (app, plugin, etc) to all supported languages
+  const projectLanguages: Record<string, string[]> = {};
+
+  // Define the commands to run for each language
+  const languageTestCommands: Record<string, Record<string, string>> = {
+    typescript: { build: "yarn build", test: "yarn test" },
+    "typescript-node": {
+      build: "yarn build",
+      test: "yarn test",
+    },
+    "typescript-react": {
+      build: "CI=false yarn build",
+    },
+    assemblyscript: {
+      build: "yarn build",
+      test: "yarn test:e2e",
+    },
+    rust: {
+      build: "yarn build",
+      test: "yarn test:e2e",
+    },
+    interface: { build: "yarn build" },
+    docusaurus: {
+      install: "yarn install --no-lockfile",
+      build: "yarn build",
+    },
+  };
+
+  // Filter unnecessary directories
+  const filter = ["node_modules", "coverage"];
+
+  // Populate all project types & languages by recursing 2 levels of directories
+  fs.readdirSync(rootDir, { withFileTypes: true })
+    .filter(
+      (dirent) => dirent.isDirectory() && filter.indexOf(dirent.name) === -1
+    )
+    .map((dirent) => dirent.name)
+    .forEach(
+      (projectType) =>
+        (projectLanguages[projectType] = fs
+          .readdirSync(path.join(rootDir, projectType), { withFileTypes: true })
+          .filter((dirent) => dirent.isDirectory())
+          .map((dirent) => dirent.name))
+    );
+
+  // for each project + language
+  for (const projectType of Object.keys(projectLanguages)) {
+    for (const language of projectLanguages[projectType]) {
+      // run the test commands at commands[language]
+      let commands = languageTestCommands[language];
+
+      // if nothing exists, try to match language as a substring
+      if (!commands) {
+        for (const testLanguage of Object.keys(languageTestCommands)) {
```

Comment:
Instead of this pattern:
```ts
for (x) {
  for (y) {
   test (`${x} {y}`) {
}
}
}
```

use following for better test hierarchy:
```ts
for (x) {
 test(x) {
  for(y) {
   test(y) {}
}
}
}
```